src/pattern_matching/myers/builder.rs

```diff
@@ -0,0 +1,179 @@
+use super::*;
+
+/// Builds a Myers instance, allowing to specify ambiguities.
+///
+/// # Example:
+///
+/// This example shows how recognition of IUPAC ambiguities in patterns
+/// can be implemented. Ambiguities in the searched text are not recognized; this
+/// would require specifying additional ambiguities (A = MRWVHDN, etc...).
+///
+/// ```
+/// # extern crate bio;
+/// use bio::pattern_matching::myers::MyersBuilder;
+///
+/// # fn main() {
+/// let ambigs = [
+///     (b'M', &b"ACM"[..]),
+///     (b'R', &b"AGR"[..]),
+///     (b'W', &b"ATW"[..]),
+///     (b'S', &b"CGS"[..]),
+///     (b'Y', &b"CTY"[..]),
+///     (b'K', &b"GTK"[..]),
+///     (b'V', &b"ACGMRSV"[..]),
+///     (b'H', &b"ACTMWYH"[..]),
+///     (b'D', &b"AGTRWKD"[..]),
+///     (b'B', &b"CGTSYKB"[..]),
+///     (b'N', &b"ACGTMRWSYKVHDBN"[..])
+/// ];
+///
+/// let mut builder = MyersBuilder::new();
+///
+/// for &(base, equivalents) in &ambigs {
+///     builder.ambig(base, equivalents);
```

Comment:
I would probably write this directly in this form:
```
builder.ambig(b'M', b"ACM");
builder.ambig(b'R', b"AGR");
```

Also, it feels unintuitive that the letter itself is repeated in the pattern.

Reply:
I don't entirely get the above point. Do you mean the usage of `&b"ACM"[..]`? It's not possible to do it using simply `b"ACM"` since these byte arrays are of different length.
About repeating the letter itself in the pattern: agreed! I changed this behavior.